src/coreclr/src/vm/eventtrace.cpp

```diff
@@ -4272,6 +4269,18 @@ VOID EtwCallbackCommon(
         ctxToUpdate->EventPipeProvider.EnabledKeywordsBitmask = MatchAnyKeyword;
     }
 
+    if ((ControlCode == EVENT_CONTROL_CODE_ENABLE_PROVIDER || ControlCode == EVENT_CONTROL_CODE_DISABLE_PROVIDER) &&
```

Comment:
I think you probably want to also maintain the `g_fEEStarted && !g_fEEShutdown` conditions here as well.

Reply:
Probably not because we really want the settings on the GC side to mimic what's set on the VM side, and the VM side does *not* check g_fEEStarted and !g_fEEShutdown before setting level and keywords.

Also, I noticed that the first call to EtwCallbackCommon happens when g_fEEStarted is still FALSE, so we'd miss that first call.

Reply:
I see.  As long as the call is safe to make during startup and shutdown, then that's OK.  I know that a bunch of the other calls that are inside the if statement below are unsafe and will throw.